etc/picongpu/hypnos-hzdr/k80_restart.tpl

```diff
@@ -99,30 +99,60 @@ do
         file=""
     fi
 done
+export file
 
 #this sed call extracts the final simulation step from the cfg (assuming a standard cfg)
-finalStep=`echo !TBG_programParams | sed 's/.*-s[[:blank:]]\+\([0-9]\+[^\s]\).*/\1/'`
+export finalStep=`echo !TBG_programParams | sed 's/.*-s[[:blank:]]\+\([0-9,:,\,]\+[^\s]\).*/\1/'`
 echo "final step      = " $finalStep | tee -a output
 #this sed call extracts the -s and --checkpoint flags
 programParams=`echo !TBG_programParams | sed 's/-s[[:blank:]]\+[0-9]\+[^\s]//g' | sed 's/--checkpoint\.period[[:blank:]]\+[0-9]\+[^\s]//g'`
 #extract restart period
-restartPeriod=`echo !TBG_programParams | sed 's/.*--checkpoint\.period[[:blank:]]\+\([0-9]\+[^\s]\).*/\1/'`
+export restartPeriod=`echo !TBG_programParams | sed 's/.*--checkpoint\.period[[:blank:]]\+\([0-9,:,\,]\+[^\s]\).*/\1/'`
 echo  "restart period = " $restartPeriod | tee -a output
 
-if [ "" != "$file" ]
-then
-    cptimestep=`basename $file | sed 's/checkpoint_//g' | sed 's/.'$fileEnding'//g'`
-    echo "start time      = " $cptimestep | tee -a output
-
-    endTime="$(($cptimestep + $restartPeriod ))"
-    echo "end time        = " $endTime | tee -a output
-
-    stepSetup=$(echo -s $endTime "--checkpoint.restart --checkpoint.restart.step" $cptimestep "--checkpoint.period" $restartPeriod )
-else
-    echo "no checkpoint found" | tee -a output
-    endTime=$restartPeriod
-    stepSetup=$(echo " -s " $endTime "--checkpoint.period" $restartPeriod )
-fi
+# ******************************************* #
+# need some magick, if the restart period is in new notation with the ':' and ','
+
+which python3 >&2
+python3 --version >&2 
```

Comment:
Sry but we should not  pull python into the templates because than you need also the correct python environment.

Reply:
well, but it works :)
3 arguments:
 - if you like, we can add a module load
 - it's only k80, so none except us will have trouble if something would not work (and we are all pros and will understand ;))
 - if it doesn't work, you see it on submitting, at once - no hidden errors, no hours of lost computing time

It would probably be cleaner to put the code in a script, that accepts some parameters and returns some, shall I? and where shall the script be located?